### ISSUE TITLE
Regexp accepted by ruby but rejected by regexp_parser

### ISSUE THREAD

**Issue by mbj** [ammar/regexp_parser#87]
```
$ bundle exec irb -r regexp_parser
irb(main):001:0> Regexp::Parser.parse("\\99")
/home/mutant-dev/.gem/ruby/3.2.2/gems/regexp_parser-2.8.1/lib/regexp_parser/parser.rb:592:in `block in assign_referenced_expressions': Invalid reference 9 at pos 0 (Regexp::Parser::ParserError)
	from /home/mutant-dev/.gem/ruby/3.2.2/gems/regexp_parser-2.8.1/lib/regexp_parser/parser.rb:590:in `each'
	from /home/mutant-dev/.gem/ruby/3.2.2/gems/regexp_parser-2.8.1/lib/regexp_parser/parser.rb:590:in `assign_referenced_expressions'
	from /home/mutant-dev/.gem/ruby/3.2.2/gems/regexp_parser-2.8.1/lib/regexp_parser/parser.rb:45:in `parse'
	from /home/mutant-dev/.gem/ruby/3.2.2/gems/regexp_parser-2.8.1/lib/regexp_parser/parser.rb:22:in `parse'
	from (irb):1:in `<main>'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/gems/3.2.0/gems/irb-1.6.2/exe/irb:11:in `<top (required)>'
	from /home/mutant-dev/.rubies/ruby-3.2.2/bin/irb:25:in `load'
	from /home/mutant-dev/.rubies/ruby-3.2.2/bin/irb:25:in `<top (required)>'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/3.2.0/bundler/cli/exec.rb:58:in `load'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/3.2.0/bundler/cli/exec.rb:58:in `kernel_load'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/3.2.0/bundler/cli/exec.rb:23:in `run'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/3.2.0/bundler/cli.rb:492:in `exec'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/3.2.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/3.2.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/3.2.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /home/mutant-dev/.rubies/ruby-3.2.2/lib/ruby/3.2.0/bundler/cli.rb:34:in `dispatch'
	... 7 levels...
irb(main):002:0> /\99/
=> /\99/
```

I suspect that `regexp_parser` has to be as lenient when it comes to backrefs as ruby itself?

**Comment by jaynetics** [comment]
@mbj

thank you for the report! i've just released v2.8.2 with a fix. there are more details in the commit message if you're interested. that was a nice exotic quirk you found there...

**Comment by mbj** [comment]
> thank you for the report! i've just released v2.8.2 with a fix. there are more details in the commit message if you're interested. that was a nice exotic quirk you found there...

mutant found it when I updated the rubyspec corpus test to latest rubyspec ref ;)
